esi_ui/api/esi_api.py

```diff
@@ -0,0 +1,67 @@
+from django.conf import settings
+
+from ironicclient.v1.client import Client as _Ironicclient
+from esi import connection
+
+from openstack import config
+from openstack_dashboard.api import base
+from horizon.utils.memoized import memoized  # noqa
+
+DEFAULT_OPENSTACK_KEYSTONE_URL = 'http://127.0.0.1/identity/v3'
+
+@memoized
+def get_session_from_token(token):
+    auth_url = getattr(settings, 'OPENSTACK_KEYSTONE_URL', DEFAULT_OPENSTACK_KEYSTONE_URL)
+    region = config.get_cloud_region(load_yaml_config=False,
+                                     load_envvars=False,
+                                     auth_type='token',
+                                     token=token,
+                                     auth_url=auth_url,
+                                     )
+    return region.get_session()
+
+@memoized
+def esiclient(token):
+    return connection.ESIConnection(session=get_session_from_token(token))
+
+@memoized
+def ironicclient(token):
+    return _Ironicclient(session=get_session_from_token(token))
+
+
+def node_list(request):
+    token = request.user.token.id
+
+    esi_nodes = esiclient(token).list_nodes()
```

Comment:
I'd suggest doing something like https://github.com/CCI-MOC/python-esiclient/blob/master/esiclient/v1/node_network.py#L70 to run these API commands in parallel - that should speed things up a bit!

Reply:
completed